easybuild/easyconfigs/p/PGI/PGI-18.7-GCC-7.3.0-2.30.eb

```diff
@@ -0,0 +1,25 @@
+name = 'PGI'
+version = '18.7'
+
+homepage = 'http://www.pgroup.com/'
+description = "C, C++ and Fortran compilers from The Portland Group - PGI"
+
+toolchain = {'name': 'dummy', 'version': ''}
+
+sources = ['pgilinux-20%(version_major)s-%(version_major)s%(version_minor)s-x86-64.tar.gz']
+checksums = ['d97225585127b841075576859fc7782c9dbe3f7e1569bbce5f465046d4679d8d']
+
+gccver = '7.3.0'
+binutilsver = '2.30'
+versionsuffix = '-GCC-%s-%s' % (gccver, binutilsver)
+
+dependencies = [
+    ('GCCcore', gccver),
+    ('binutils', binutilsver, '', ('GCCcore', gccver)),
+    ('numactl', '2.0.11', '', ('GCCcore', gccver)),
```

Comment:
Probably need to drop numactl here.
current hwloc need numactl 2.0.11 and not the one that PGI ships.
And if numactl is dep for PGI, numactl will get loaded before PGI, and thus hwloc will see PGI's too old version.

I ran into this yesterday.

Reply:
How does that issue manifest itself?

If we don't include `numactl` as a dep, anything that is built with PGI without having `numactl` listed as a dependency would be using the old version included with PGI though...

Reply:
Yes, that is a potential problem.
Trying to build hwloc 1.11.10 with this PGI fails, i.e. the poolf-2018b I'm trying to finish.

The real problem is something i had a solution for way back, pre EB.
The PGI installer doesn't properly realize that there is an installed numa library.
Something they seem to have broken again, between 17.10 and 18.7.

This despite that we have libnuma installed from the OS too.

So, it creates a libnuma -> libpgnuma softlink (for all the .a/.so stuff)
This breaks things.

I'll take a look and see if i can figure out how they have broken things in their installer, or if we simply need to eliminate those bad links (i.e., what we used to do).

Reply:
Ah, first of all the makelocalrc is still broken for Ubuntu multiarch systems.

Secondly, the code in pgi.py for removing the bad libnuma softlinks seems to not work for some reason. I'll investigate.

Reply:
Why does
if get_software_root("numactl"):
fail when this in the easyconfig
```
dependencies = [
    ('GCCcore', gccver),
    ('binutils', binutilsver, '', ('GCCcore', gccver)),
    ('numactl', '2.0.11', '', ('GCCcore', gccver)),
]
```

Reply:
I.e., in pgi.py i have added
```
        print "AKE: Checking libnuma"
        if get_software_root("numactl"):
            print "AKE: numactl found"
```
and the build output is:
```
== installing...
AKE: Checking libnuma
== Running post-install hook...
```

Reply:
both
```
        print "AKE: conf_step: Checking libnuma %s" % get_software_root("numactl")
        print "AKE: conf_step: Checking GCCcore %s" % get_software_root("GCCcore")
```
says "None".

I'm too tired to figure this out right now.

Reply:
@akesandgren Make sure that the toolchain `version` is *not* set to `dummy`, see https://easybuild.readthedocs.io/en/latest/Writing_easyconfig_files.html#loading-of-modules-for-dependencies-with-a-dummy-toolchain

Reply:
Doh.... I was fiddling with that for another reason and forgot to change back...

Reply:
So the conclusion here is that I messed up. There is no problem with numactl being defined like this. At least not at the moment, though I can think of situations where it might be better to have in in builddependencies.